### ISSUE TITLE
Some whole values in tensors are stored incorrectly

### ISSUE THREAD

**Issue by ArtemShkil** [tenstorrent/tt-metal#35275]
### Component / Area

ops

### Issue Type (optional)

Bad Outputs

### Observed

In the tensors with layout Layout::TILE and data type DataType::BFLOAT16, whole numbers that are larger than 255 are usually stored incorrectly. This happens when tensors are created from PyTorch or as a result of operations.

For example, tensor [566., 567., 568., 569.] is stored as [568., 568., 568., 568.].

### Expected

Whole numbers that are larger than 255 are stored correctly.

### 1. Steps (exact commands)

```python
import torch
import ttnn

device_id = 2
device = ttnn.open_device(device_id=device_id)

torch_tensor = torch.tensor([566., 567., 568., 569.], dtype=torch.float32)
ttnn_tensor = ttnn.from_torch(torch_tensor, layout=ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16, device=device)

print(f'Value of ttnn_tensor: {ttnn_tensor}')

a_torch = torch.tensor([560., 560., 560., 560.], dtype=torch.float32)
b_torch = torch.tensor([6., 7., 8., 9.], dtype=torch.float32)

a_ttnn = ttnn.from_torch(a_torch, layout=ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16, device=device)
b_ttnn = ttnn.from_torch(b_torch, layout=ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16, device=device)
c_ttnn = a_ttnn + b_ttnn

print(f'Value of a_ttnn: {a_ttnn}')
print(f'Value of b_ttnn: {b_ttnn}')
print(f'Value of c_ttnn: {c_ttnn}')
```

Output:
```
Value of ttnn_tensor: ttnn.Tensor([568.0000, 568.0000, 568.0000, 568.0000], shape=Shape([4]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
Value of a_ttnn: ttnn.Tensor([560.0000, 560.0000, 560.0000, 560.0000], shape=Shape([4]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
Value of b_ttnn: ttnn.Tensor([ 6.0000,  7.0000,  8.0000,  9.0000], shape=Shape([4]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
Value of c_ttnn: ttnn.Tensor([568.0000, 568.0000, 568.0000, 568.0000], shape=Shape([4]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
```

### 2. Input data / link or description

PyTorch tesnor [566., 567., 568., 569.].

### 3. Frequency

Always

### 1. Software Versions

Ubuntu 22.04

### 2. Hardware Details

Wormhole N150D

### Is this a regression?

Unknown

### Priority

None

**Comment by jasondavies** [comment]
These values are not exactly representable in bfloat16: `[566., 567., 568., 569.]`.  This format has an 8-bit mantissa, including the implicit bit.  These values require 10 bits; the exponent is `9`, and so for bfloat16, the values will be `564, 568, 572, ...`.

**Comment by cmaryanTT** [comment]
Expected behaviour. Integers higher than 256 cannot be represented by bf16 exactly, as it only has 7 mantissa bits.

**Comment by ArtemShkil** [comment]
Can you please add label #EPAM_DX?
